include/bluetooth/services/ots.h

```diff
@@ -545,9 +545,9 @@ struct bt_ots_cb {
 	 *  @return Data length to be sent via data parameter. This value
 	 *          shall be smaller or equal to the len parameter.
 	 */
-	uint32_t (*obj_read)(struct bt_ots *ots, struct bt_conn *conn,
-			     uint64_t id, uint8_t **data, uint32_t len,
-			     uint32_t offset);
+	size_t (*obj_read)(struct bt_ots *ots, struct bt_conn *conn,
+			   uint64_t id, void **data, uint32_t len,
+			   uint32_t offset);
```

Comment:
Hmm, I see that the previous version simply had a `uint32_t` return value, which implies that the application cannot send any error condition back, and why `size_t` makes sense. 

To be POSIX compliant(ish), I guess we need to change this to a `ssize_t` so that the application may return with an error condition as well (that needs to be checked in `ots` when calling this callback).

Reply:
I will make a change to the read mechanism to check for errors and terminate the channel if an error condition is returned. This will make it analogous to the write path where an error other than `EINPROGRESS` terminates the channel.

Reply:
It seems a bit inconsistent to me to have a `uint32_t len` input parameter but `size_t` return value, when those are supposed to be of the same "unit", i.e. the return value tells you how many bytes between `0` and `len` was read.

Reply:
I have changed the return type to `ssize_t` and the parameters to `size_t` and `off_t`.